src/Infrastructure/CreditCardExpiry.php

```diff
@@ -22,6 +22,15 @@ public function __construct( int $month, int $year ) {
 		$this->year = $year;
 	}
 
+	public static function newFromString( string $expirationDate ) {
+		$dateParts = explode( '/', $expirationDate );
+		if ( count( $dateParts ) === 2 ) {
+			return new self( (int)$dateParts[0], (int)$dateParts[1] );
+		}
+
+		return null;
```

Comment:
We might not receive that data from the payment provider's service API. Not making it nullable means that we can never set the transaction data without it.